core/templates/dev/head/domain/collection/CollectionRightsObjectFactory.js

```diff
@@ -0,0 +1,110 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating and mutating instances of frontend
+ * collection rights domain objects.
+ */
+
+oppia.factory('CollectionRightsObjectFactory', [
+    function() {
+    var CollectionRights = function(collectionRightsObject) {
+      this._collectionId = collectionRightsObject.collection_id;
+      this._canEdit = collectionRightsObject.can_edit;
+      this._canUnpublish = collectionRightsObject.can_unpublish;
+      this._isPrivate = collectionRightsObject.is_private;
+      this._ownerNames = collectionRightsObject.owner_names;
+    };
+
+    // Instance methods
+
+    CollectionRights.prototype.getCollectionId = function() {
+      return this._collectionId;
+    };
+
+    // Returns true if the the user can edit the collection. This property is
+    // immutable.
+    CollectionRights.prototype.canEdit = function() {
+      return this._canEdit;
+    };
+
+    // Returns true if the user can unpublish the collection.
+    CollectionRights.prototype.canUnpublish = function() {
+      return this._canUnpublish;
+    };
+
+    // Returns true if the collection is private.
+    CollectionRights.prototype.isPrivate = function() {
+      return this._isPrivate;
+    };
+
+    // Returns true if the collection is public.
+    CollectionRights.prototype.isPublic = function() {
+      return !this._isPrivate;
+    };
+
+    CollectionRights.prototype.setPublic = function() {
+      if (this._canEdit) {
+        this._isPrivate = false;
+      } else {
+        throw new Error('User is not allowed to edit this collection.');
+      }
+    };
+
+    // Sets isPrivate to true only if canUnpublish in is true.
+    CollectionRights.prototype.setPrivate = function() {
+      if (this.canUnpublish()) {
+        this._isPrivate = true;
+      } else {
+        throw new Error('User is not allowed to unpublish this collection.');
+      }
+    };
+
+    // Returns the owner names of the collection. This property is immutable.
+    CollectionRights.prototype.getOwnerNames = function() {
```

Comment:
Is this going to bind correctly in Angular? It seems like a reset of the collection rights object would cause this to not be updated on the permissions card. I think you need a bindable version of this method with corresponding tests and documentation (see ``CollectionObjectFactory. getBindableCollectionNodes()``).